### ISSUE TITLE
pygame assumes that SDL minor version bump implies ABI break, but this is no longer true

### ISSUE THREAD

**Issue by smcv** [pygame/pygame#3427]
**Environment:**

- **Operating system**: Linux (Debian)
- **Python version**: 3.10
- **SDL version**: 2.24
- **PyGame version**: 2.1.2
- **Relevant hardware**: n/a

**Current behavior:**

The version of pygame currently in Debian was compiled against SDL 2.0.22. When run after upgrading SDL to 2.24.0, it crashes with this message:

```
RuntimeError: SDL compiled with version 2.0.22, linked to 2.24.0
```

**Expected behavior:**

pygame should either not second-guess the library dependency system at all, or only check the major version.

SDL 2.x has always been backwards-compatible as far as SDL 2.0.0, which made the previous interpretation of the version number fairly useless: the minor version was stuck at zero, and the micro version was used for "feature" updates, leaving no way to make bugfix-only point releases.

Since SDL 2.24.0, there is a [new versioning scheme](https://github.com/libsdl-org/SDL/blob/main/docs/README-versions.md): the minor version is used for "feature" updates and to indicate development vs. stable status, leaving the micro version available to indicate point releases. This is the same versioning scheme used by GLib, GTK, Flatpak and older versions of the Linux kernel.

**Steps to reproduce:**

1. Build pygame against an older version of SDL. The one currently in Debian testing was compiled with 2.0.22.
2. Upgrade SDL. The version currently in Debian unstable is 2.24.0.
3. Run any pygame game. The bug reporter in Debian used `bambam`.

**Test code**

```
import pygame
pygame.init()
```

**Stack trace/error output/other error logs**

```
Traceback (most recent call last):
  File "/usr/games/bambam", line 433, in <module>
    main()
  File "/usr/games/bambam", line 426, in main
    bambam.run()
  File "/usr/games/bambam", line 343, in run
    pygame.init()
RuntimeError: SDL compiled with version 2.0.22, linked to 2.24.0
```

**Comment by smcv** [comment]
The new versioning scheme was introduced in https://github.com/libsdl-org/SDL/pull/5602 to resolve the issues discussed in https://github.com/libsdl-org/SDL/issues/5217.

**Comment by ankith26** [comment]
Thanks for the issue report! 
I've put a PR to fix this, should this be considered a 2.1.3 release blocker?

**Comment by robertpfeiffer** [comment]
I am kind of confused by how this crept in.

**Comment by robertpfeiffer** [comment]
Also holy shit when did Debian upgrade from 1.9.4?

**Comment by smcv** [comment]
> Also holy shit when did Debian upgrade from 1.9.4?

January 2020, according to <https://tracker.debian.org/pkg/pygame>. Debian 11 has 1.9.6, and that will not change, because the purpose of Debian stable releases is to be a fixed point without behaviour changes or regressions (even if that means some bugs not being fixed). The testing/unstable branch that will become Debian 12 currently has 2.1.2.

Debian 10 is still on 1.9.4, but Debian 10 has reached EOL and is no longer maintained by the project (the Debian LTS subproject is still applying security fixes to a subset of it, but they'd probably ignore pygame as outside their scope).

**Comment by smcv** [comment]
> I wrote the check with SemVer in mind

SDL's documented versioning scheme isn't *exactly* semver, but it's close enough. (Stable releases 2.even.z are basically semver, but development prereleases 2.odd.z don't guarantee backwards-compatibility.)

Before this change, SDL's versioning scheme wasn't really documented: the SDL major version behaved like a semver major version, the SDL micro version (the z in x.y.z) behaved like a semver minor version (the y in x.y.z), nobody really knew what a nonzero SDL minor version would mean, and there was no equivalent of a semver micro version.

**Comment by MyreMylar** [comment]
> Thanks for the issue report! I've put a PR to fix this, should this be considered a 2.1.3 release blocker?

I think we should get this fixed now. I noticed a user complaint about this on reddit earlier too.

**Comment by illume** [comment]
> I think we should get this fixed now. I noticed a user complaint about this on reddit earlier too.

Thanks to Ankith and co... the fix is merged in now. https://github.com/pygame/pygame/pull/3428
